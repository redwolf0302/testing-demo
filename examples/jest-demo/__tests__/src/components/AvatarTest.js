import React from 'react';
import renderer from 'react-test-renderer';

describe('Avatar Correctly', () => {
    const Avatar = require('../../../dist/src/components/Avatar/Avatar').default;
    test('Avatar snapshot', () => {
        const tree = renderer.create(<Avatar width={30} height={30}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Avatar snapshot that has class name \'test\'', () => {
        const tree = renderer.create(<Avatar width={30} height={30} cls={'test'}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Avatar snapshot with imageUrl', () => {
        const tree = renderer.create(<Avatar width={30} height={30}
                                             imageUrl={'https://static.jk.cn/XXX.jpg'}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});