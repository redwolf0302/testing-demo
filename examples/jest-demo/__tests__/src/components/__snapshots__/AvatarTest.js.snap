// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Correctly Avatar snapshot 1`] = `
<div
  className="es-avatar"
  onClick={[Function]}
  style={
    Object {
      "height": 30,
      "minHeight": 30,
      "minWidth": 30,
      "width": 30,
    }
  }
>
  <img
    className="es-avatar-img"
    onError={[Function]}
    onLoad={[Function]}
  />
</div>
`;

exports[`Avatar Correctly Avatar snapshot that has class name 'test' 1`] = `
<div
  className="es-avatar test"
  onClick={[Function]}
  style={
    Object {
      "height": 30,
      "minHeight": 30,
      "minWidth": 30,
      "width": 30,
    }
  }
>
  <img
    className="es-avatar-img"
    onError={[Function]}
    onLoad={[Function]}
  />
</div>
`;
exports[`Avatar Correctly Avatar snapshot with imageUrl 1`] = `
<div
  className="es-avatar"
  onClick={[Function]}
  style={
    Object {
      "height": 30,
      "minHeight": 30,
      "minWidth": 30,
      "width": 30,
    }
  }
>
  <img
    className="es-avatar-img"
    onError={[Function]}
    onLoad={[Function]}
    src="https://static.jk.cn/XXX.jpg"
  />
</div>
`;
